### DISCUSSION TITLE
Run Time Update

### DISCUSSION THREAD

**Discussion by satzguyz** [nasa/fprime#1361] [Q&A]
Hello,

Is it possible to update the components during Run-time of F-Prime SW while in satellite on-board.

**Comment by LeStarch** [comment]
Do you mean software updating, e.g. send new software, reboot into new software?  This is possible, but left to projects to implement within F´ components as software images, disk, etc are project specific.

Alternatively, do you mean dynamic reconfiguration of an F´  software deployment?  Connections in F´ are created at during initialization of software (runtime), and theoretically could be changed later during the execution of the software. However, I don't know of any projects doing this.

**Reply by satzguyz** [reply]
@LeStarch I mean the updating the specific component alone. i.e instead of uploading new software and reboot, is there options to update specific component (say specific service) alone ?
